### ISSUE TITLE
Difference unleash and yggdrasil (edge) Frontend api having an inverted constrain with missing context field

### ISSUE THREAD

**Issue by Iandenh** [Unleash/yggdrasil#136]
### Describe the bug

When trying stuff out, I found a difference between how Unleash handles a scenario and Yggdrasil (I also reproduced it in Unleash Edge). When you create an inverted IN check, and you don't push the context field you get feature enabled back from unleash (what I was expecting), but yggdrasil decides to put it as disabled,

### Steps to reproduce the bug

1. Create a feature flag with a strategy flexible rollout
2. Add a constrain with inverted IN with a value
3. Request with the frontend API the feature flag without sending the context field

### Expected behavior

That the feature is enabled. 

When using NOT_IN instead of an inverted IN, it works correctly.

I also checked it quickly with NUM_EQ, and there is the same problem. Unleash enables the feature yggdrasil not. And with this one, we don't have a native numeric not equals operator.

### Screenshots

![image](https://github.com/user-attachments/assets/c26889bc-b7c0-4cfe-9c8b-a269e52be0ee)

### Additional context

Versions:
Unleash edge build locally from this commit sha: 933659b0461b8a53cb89caf0984963918fe82b04

Same result with custom context fields and the ones already provided like appName.

### Unleash version

6.0.6

### Subscription type

None

### Hosting type

Self-hosted

**Comment by daveleek** [comment]
Hello @Iandenh, thank you for raising this and providing us with a PR.
After a little experimentation we found that we have some inconsistencies in our SDKs around how they handle this. It's an edge case that we don't have good specifications for, so we wanted to make some decisions and standardize this across all our SDKs so they behave the same way.

Yggdrasil and its behaviour, intended as a backbone for SDKs will be "the source of truth" for our SDKs going forwards, so there may be some change in behaviour in other SDKs.

For this specific case the behaviour we decided on is that a missing context field will yield `false`, but when that constraint is inverted it will result in `true`.
We decided to move the solution a little further out so we could address all of the context field types the same way in this [PR](https://github.com/Unleash/yggdrasil/pull/139). 

We're going to close your PR and merge #139, but will keep this issue open a little while longer.
We'll release an update as well.

Thank you for your contribution!
